File: Basic/Day-04/LinearSearch01.py
n = int(input("Enter number of elements you want in the list : "))
nums = []
print("Enter the elements of the list :-\n")
for i in range(n):
    nums.append(int(input(f"Enter the element {i+1} : ")))

target =int(input("Enter the number you want to search in the list : "))


def linearSearch(nums,target):
    for i in range(len(nums)):
        if nums[i]==target:
            return i

    return -1

result = linearSearch(nums,target)
if result == -1:
    print("Not Found")
else:
    print(f"{target} found at index : {result}\nposition : {result+1}")

File: BasicToAdvance/Java/Day-04/LinearSearch01.java
import java.util.Scanner;
import java.util.ArrayList;

public class LinearSearch01 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        ArrayList<Integer> list = new ArrayList<>();
        System.out.print("Enter number of elements: ");
        int n = sc.nextInt();
        for (int i = 0; i < n; i++) {
            System.out.print("Enter element " + (i + 1) + ": ");
            list.add(sc.nextInt());
        }
        System.out.print("Enter target element to search: ");
        int target = sc.nextInt();
        int result = linearSearch(list, target);
        if (result == -1) {
            System.out.println("Element not found");
        } else {
            System.out.println("Element found at index: " + result);
        }
        sc.close();
    }

    public static int linearSearch(ArrayList<Integer> list, int target) {
        for (int i = 0; i < list.size(); i++) {
            if (list.get(i) == target) {
                return i;
            }
        }
        return -1;
    }
}